scipy/__init__.py

```diff
@@ -139,7 +139,7 @@ def __dir__():
     # In maintenance branch, change to np_maxversion N+3 if numpy is at N
     # See setup.py for more details
     np_minversion = '1.16.5'
-    np_maxversion = '9.9.99'
+    np_maxversion = '1.23.0'
```

Comment:
This is right; if `1.21.0` is released before our last 1.7.0 RC then we could consider bumping this to `1.24.0`